app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -158,25 +160,30 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
 
     private fun getChangeListenerForDeviceName(): Preference.OnPreferenceChangeListener {
         return Preference.OnPreferenceChangeListener { _, newValue ->
-            // Optimistically set the device name to what user requested.
-            val deviceNameKey = context!!.getPreferenceKey(R.string.pref_key_sync_device_name)
-            val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
-            preferenceDeviceName?.summary = newValue as String
-
-            // This may fail, and we'll have a disparity in the UI until `updateDeviceName` is called.
-            lifecycleScope.launch(IO) {
-                try {
-                    accountManager.authenticatedAccount()?.let {
-                        it.deviceConstellation().setDeviceNameAsync(newValue)
+            // The network request requires a nonempty string, so don't persist any changes if the user inputs one.
+            if (newValue.toString().trim() == "") {
+                Toast.makeText(context, "Device name cannot be empty.", Toast.LENGTH_LONG).show()
+                false
+            } else {
```

Comment:
Instead of wrapping the rest of this conditional in an `else` statement, it may be better to just do an early return. (Like how you have with the false). If it's still running past this code you may need to do `return@OnPreferenceChangeListener false`.